modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java

```diff
@@ -1223,7 +1224,22 @@ else if (Objects.equals(clientExtension.type, "siteInitializer")) {
 		}
 		else if (Objects.equals(clientExtension.type, "themeCSS")) {
 			_validateTypeSettingsValues(
-				clientExtension, "scope", "controlPanel", "page");
+				clientExtension, "scope", "controlPanel", "layout");
+		}
+	}
+
+	private void _validateGlobalJSInstanceScope(
+		ClientExtension clientExtension) {
+
+		Map<String, Object> typeSettings = clientExtension.typeSettings;
+
+		String scope = String.valueOf(typeSettings.get("scope"));
+
+		if ((scope != null) && scope.equalsIgnoreCase("instance")) {
+			throw new GradleException(
+				"GlobalJS client extension " + clientExtension.id + " no " +
+					"longer has 'instance' as a valid value for the 'scope' " +
+						"attribute. Use 'company' instead.");
```

Comment:
@drewbrokke done, is it good? :eyes:

Reply:
@thiago-buarqque after looking at this a bit more, I wonder if this is even necessary. Having `_validateTypeSettingsValues` should be enough already. Thoughts?

Reply:
In theory yes, but that's not so user friendly. The dev would be like "what, instance used to work"

Reply:
At least this gives a explicit error message

Reply:
That makes sense, and I am generally a fan of explicit error messages. But in this case, the feature is relatively new, and the message from the other exception will be "helpful enough" to guide the dev through using it. Once we have a proper breaking change procedure for non-java changes like this, we should apply it, but I think for now we can reasonably omit this one.

Reply:
Also, are you planning to backport the DXP changes to older quarterly branches? If a dev uses a new version of workspace that requires "company" and targets an older version of DXP that expects "instance", the feature will not work, right?

Reply:
I see that 6fdd4f3d24db0 keep backwards compatibility for new versions, but I think the scenario from my previous comment may be valid still.

If I'm thinking about this right, we may want to keep backwards compatibility in our tooling as well. It would be easy enough to add a warning when using actually deprecated values, but still allow them.

Thoughts?

Reply:
>If I'm thinking about this right, we may want to keep backwards compatibility in our tooling as well. It would be easy enough to add a warning when using actually deprecated values, but still allow them.

@drewbrokke  Are you suggesting to just add a warning in the Gradle workspace instead of failing? If so, the warning could mention the last version in which the "instance" value is valid.

cc: @ethib137

Reply:
Yes that is possible. For this case, since the feature itself is so new and adoption is probably low, we could get away with just removing, but in the future we may want to allow some kind of backwards compatibility for a time.

In any case, we should omit this special one-off validation in this instance.

Reply:
Cool. So we go with the warning, right?

Reply:
No, we just remove and require the new value since this feature is so new. If needed, we can backport the compatibility change to older releases of DXP.

I am suggesting that we do the deprecation warning in the case where a feature is more widely used. Sorry if I mis-spoke.

Reply:
I see. So we're keeping the code the way it is.

Reply:
One thing I just noticed. There isn't an official documentation for the previous GlobalJS feature. The [PR for the docs](https://github.com/rbohl/liferay-learn/pull/158) is still open. Maybe this could help us?